Show Alert message in Facility Locator when the vets-api returns a non-200 response

mpelzsherman opened the pull request:
Resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/29076

Displays an error alert when the Facility Locator receives a bad (non-200) API response.

Note: Use of the `MockLocatorApi ` in e2e tests was preventing us from mocking the error response in Cypress using `cy.intercept`. It also added a bunch of test-specific code, so it prevented the e2e tests from actually testing the "real" API code. So I've deleted it! 

We should be using `mockFetch` in our unit tests as needed, but none of our unit tests actually trigger any fetches, so we're good.

mpelzsherman committed:
Show a system error when error received and no results found. (WIP/MOCKUP!)

mpelzsherman committed:
Show Alert message in Facility Locator when the vets-api returns a non-200 response

mpelzsherman committed:
Fix SearchResultMessage unit tests

mpelzsherman committed:
Do not display search results header if error is present.

Change alert type to warning.

Remove left padding from search results container.

mpelzsherman commented:
Thanks @ndsprinkle but this PR is still in draft. You may want to re-review after I've marked it ready for review.

ndsprinkle commented:
@mpelzsherman Ah sorry about that, wasn't paying attention. I will be sure to re-review later!

mpelzsherman committed:
Added server error test.

Changed Cypress tests to use the "real" LocatorAPI, which can be mocked to return server errors.

mpelzsherman committed:
Fixed failures in errorMessages.cypress.spec.js

Also fixed a11y issue found by that test in CCProviderPhoneLink

mpelzsherman committed:
Fixed failures in facilitySearch.cypress.spec.js

Always show 911 banner for emergency care searches even on error

More ally fixes and mock data adjustments

mpelzsherman committed:
Fixed failures in map-zoom.cypress.spec.js

Don't show error alert for mapbox errors

mpelzsherman committed:
Fixed failures in mobile.cypress.spec.js and providerSearch.cypress.spec.js

mpelzsherman commented on src/applications/facility-locator/api/index.js:
```diff
@@ -3,11 +3,7 @@ import LiveApi from './LocatorApi';
-  if (isUnitTest || isCypress) return MockApi;
-
-  return LiveApi;
+  return isUnitTest ? MockApi : LiveApi;
```
Use of the `MockLocatorApi ` in e2e tests was preventing us from mocking the error response in Cypress using `cy.intercept`. It also adds a bunch of test-specific code, so it prevents the e2e tests from actually using the "real" API code. We should be using `mockFetch` in our unit tests as needed, and get rid of trhe `MockLocatorApi` altogether ASAP.

mpelzsherman committed:
Removed MockLocatorApi

kelsonic reviewed (approved):
lgtm!
